brion/plugin/compartmentReportBinary.cpp

```diff
@@ -251,16 +348,101 @@ floatsPtr CompartmentReportBinary::loadFrame(const float timestamp) const
             const uint64_t targetOffset = offsets[i][j];
 
             for (uint16_t k = 0; k < numCompartments; ++k)
-                (*buffer)[k + targetOffset] = source[k + sourceOffset];
+                buffer[k + targetOffset] = source[k + sourceOffset];
         }
     }
 
     if (_header.byteswap)
     {
 #pragma omp parallel for
         for (ssize_t i = 0; i < ssize_t(_subNumCompartments); ++i)
-            lunchbox::byteswap((*buffer)[i]);
+            lunchbox::byteswap(buffer[i]);
+    }
+    return true;
+}
+
+bool CompartmentReportBinary::_loadFrameAIO(const float timestamp,
+                                            float* buffer) const
+{
+    if (_offsets[_subtarget].empty())
+        return false;
+
+    const size_t frameNumber = _getFrameNumber(timestamp);
+    const size_t frameOffset =
+        _header.dataBlockOffset +
+        _header.numCompartments * sizeof(float) * frameNumber;
+
+    if (!_subtarget)
+    {
+        // clang-format off
+        auto cb = readAsync(
+            {
+                _fileDescriptor,
+                buffer,
+                _header.numCompartments * sizeof(float),
+                frameOffset
+            }
```

Comment:
This block confused me a little bit at first sight, I thought it was a lambda. I prefer initializer lists to be styled the same as function calls.